content/1_docs/2_cookbook/7_extensions/0_kirby-loves-cdn/recipe.txt

```diff
@@ -0,0 +1,266 @@
+
+Push Zones are often recommended or even required if you want to host files beyond a certain size.
+
+## Pull Zones
+
+A Pull Zone is the most common Zone type and easier to set up. With this method, the CDN automatically pulls content from your web server and caches it on the CDN's servers. Once the content is cached, visitor requests will be routed and delivered from the nearest possible server location.
+
+So for example if your logo is located at
+
+```
+https://yourdomain.com/assets/images/logo.png
+```
+
+it will be requested from your Pull Zone like this:
+
+```
+https://yourpullzone.yourcdn.com/assets/images/logo.png
+```
+
+The Pull Zone will fetch the logo on the first request and spread it across the network.
+
+In this recipe, we will look at using a Pull Zone approach with (link: https://www.keycdn.com/ text: KeyCDN), the CDN provider we also use on the Kirby website. However, this approach should also work similarly with other CDN providers.
+
+## Setting up your Pull Zone
+
+The first step is to create a Pull Zone in your CDN provider's account settings. For KeyCDN you can find the instructions in their documentation in the chapter (link: https://www.keycdn.com/support/create-a-pull-zone text: Create a Pull Zone).
+
+## Zone Alias
+
+Optionally, you can create a Zone Alias. Zone Aliases allow you to use your own custom CDN URL (e.g. `cdn.yourdomain.com`) instead of the KeyCDN URL. For this to work, you have to add the Zone Alias as CNAME record in your DNS.
+
+Head over to (link: https://www.keycdn.com/support/create-a-zone-alias text: Create a Zone Alias) if you want to do this and follow the instructions.
+
+<info>
+If your website uses TLS, set up a certificate first before you create a Zone Alias.
+</info>
+
+Once you have set up your Zone and an optional Zone Alias, you are ready to start working on the Kirby side of things.
+
+## KeyCDN plugin
+
+On the Kirby side, we now have to make sure that the assets we want to serve from the CDN use the CDN's Zone URL instead of the ones linked to the website's domain, and this without us having to manually change each URL. We can achieve this with the (link: docs/reference/plugins/components/url text: `url` component extension), which will take care of rerouting the relevant paths.
+
+On a Kirby site there are typically two folders with files for which we would need to change the URL to point to a CDN:
+
+- assets
+- media
+
+<info>
+If you use a different folder where you store your assets, you will have to change the path accordingly.
+</info>
+
+### Setting the CDN domain in `config.php`
+
+Let's start with configuring the options for the CDN plugin in `config.php` (if this file doesn't exist, create it in `/site/config`):
+
+```php "/site/config/config.php"
+<?php
+return [
+  'keycdn'        => true,
+  'keycdn.domain' => 'https://yourpullzone.yourcdn.com'
```

Comment:
I'd rename the option and plugin names to `cdn` as they are not specific to KeyCDN. Otherwise it could look like an ad, which it isn't.

Also applies to the following code below.